qos-monitor/src/main/java/eu/arrowhead/core/qos/dto/externalMonitor/ExternalMonitorOrchestrationRequestFactory.java

```diff
@@ -75,16 +77,18 @@ public SystemRequestDTO createExternalMonitorSystemRequestDTO() {
 
 		final PublicKey publicKey = (PublicKey) arrowheadContext.get(CommonConstants.SERVER_PUBLIC_KEY);
```

Comment:
You should move this line inside the if block. Or even better if you move inside the sslProperties.isSslEnabled() if block.

Reply:
done